Description: Use flags set by dpkg-buildflags to enable automatic hardening
Author: Ludovic Rousseau <rousseau@debian.org>
--- a/makefile
+++ b/makefile
@@ -1,3 +1,7 @@
+CPPFLAGS:=$(shell dpkg-buildflags --get CPPFLAGS)
+CFLAGS:=$(shell dpkg-buildflags --get CFLAGS)
+LDFLAGS:=$(shell dpkg-buildflags --get LDFLAGS)
+
 #--------------------------------
 # jhead makefile for Unix
 #--------------------------------
@@ -11,10 +15,10 @@
 	$(OBJ)/exif.o $(OBJ)/iptc.o $(OBJ)/gpsinfo.o $(OBJ)/makernote.o 
 
 $(OBJ)/%.o:$(SRC)/%.c
-	${CC} $(CFLAGS) -c $< -o $@
+	${CC} $(CFLAGS) $(CPPFLAGS) -c $< -o $@
 
 jhead: $(objs) jhead.h
-	${CC} -o jhead $(objs) -lm
+	${CC} $(LDFLAGS) -o jhead $(objs) -lm
 
 clean:
 	rm -f $(objs) jhead
